tests/org.palladiosimulator.dataflow.confidentiality.analysis.tests/src/org/palladiosimulator/dataflow/confidentiality/analysis/DataflowSequenceFinderTest.java

```diff
@@ -0,0 +1,155 @@
+import static org.palladiosimulator.dataflow.confidentiality.analysis.AnalysisUtils.assertSequenceElements;
+import static org.palladiosimulator.dataflow.confidentiality.analysis.AnalysisUtils.assertUserSequenceElementContent;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.palladiosimulator.dataflow.confidentiality.analysis.sequence.entity.pcm.CallingSEFFActionSequenceElement;
+import org.palladiosimulator.dataflow.confidentiality.analysis.sequence.entity.pcm.CallingUserActionSequenceElement;
+import org.palladiosimulator.dataflow.confidentiality.analysis.sequence.entity.pcm.SEFFActionSequenceElement;
+
+public class DataflowSequenceFinderTest extends AnalysisFeatureTest {
+
+    /**
+     * Tests whether the analysis finds the correct amount of sequences
+     * 
+     * @param analysis
+     *            Analysis model that should be tested
+     * @param expectedSequences
+     *            Expected number of sequences
+     */
+    @ParameterizedTest
+    @DisplayName("Should return the correct number of sequences")
+    @MethodSource("testCountProvider")
+    public void testCount(StandalonePCMDataFlowConfidentialtyAnalysis analysis, int expectedSequences) {
+        var allSequences = analysis.findAllSequences();
+        assertEquals(expectedSequences, allSequences.size(),
+                String.format("Expected two dataflow sequences, but found %s sequences", allSequences.size()));
+    }
+
+    /**
+     * Provides a list of arguments to the {@code testCount} Test. The list contains the arguments
+     * to the test method
+     * 
+     * @return List of arguments to the test method
+     */
+    private static Stream<Arguments> testCountProvider() {
+        return Stream.of(Arguments.of(onlineShopAnalysis, 2), Arguments.of(internationalOnlineShopAnalysis, 1),
+                Arguments.of(travelPlannerAnalysis, 2));
+    }
+
+    @ParameterizedTest
+    @DisplayName("Should return the correct sequence of Dataflow Variables")
+    @MethodSource("testPathProvider")
+    public void testPath(StandalonePCMDataFlowConfidentialtyAnalysis analysis, List<List<Class<?>>> classes) {
+        var sequences = analysis.findAllSequences();
+
+        assertTrue(sequences.size() >= classes.size());
+
+        for (int i = 0; i < classes.size(); i++) {
+            assertSequenceElements(sequences.get(i), classes.get(i));
+        }
+    }
+
+    /**
+     * Provides a list of arguments to the {@code testPath} Test. The list contains the arguments to
+     * the test method
+     * 
+     * @return List of arguments to the test method
+     */
+    private static Stream<Arguments> testPathProvider() {
+        return Stream.of(
+                Arguments.of(onlineShopAnalysis,
```

Comment:
This is a hard to read and to maintain. I would propose to save the lists of element classes in properly named constants and then only referring to them here.

Reply:
I agree. I`ve put the Paths for the test models into a separate class and changed the formatting to make it more clear, what data the Lists provide. Furthermore i`ve updated the documentation for the argument provider accordingly